File: backend/app/schemas/task.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel

class TaskBase(BaseModel):
    title: str
    description: Optional[str] = None
    is_completed: bool = False

class TaskCreate(TaskBase):
    pass

class TaskRead(TaskBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime]
    user_id: int

class TaskUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    is_completed: Optional[bool] = None

File: frontend/types/task.ts
export interface Task {
  id: number;
  title: string;
  description?: string;
  is_completed: boolean;
  created_at: string;
  updated_at?: string;
  user_id: number;
}

export interface TaskFormValues {
  title: string;
  description?: string;
  is_completed?: boolean;
}

export interface PaginatedTasks {
  items: Task[];
  total: number;
  page: number;
  per_page: number;
  total_pages: number;
}

export type TaskFilter = "all" | "completed" | "pending";
